### PR TITLE
Restaking service + validator RPC improvement

### LINKED ISSUE #1121: [Docs] RPC: Block request should return the block producer's account_id
And any other relevant information.

### LINKED ISSUE #1637: Validators RPC should show details about current assignment
Currently it's unclear who is assigned to which shard and order of block production.

Also we should have just a separate end point to query validators instead of using `query`.
```
http post http://localhost:3030/ method:=validators params:="" id=dontcare jsonrpc=2.0
```

Additionally, let's surface the information about how much missing blocks / chunks each validator has.

### PR DESCRIPTION
**Pull request by ilblackdragon**
(Previous PR #1894 was based of master, redone)

This implements a separate service that run in parallel with nearcore and re-stakes if it sees that given validator might be kicked out or already kicked out.

Usage:
```
cargo build -p restaked --release
./target/release/restaked \
    --home=/path/to/validator_key.json \
    --rpc-url=http://url.to.your.node.or.any.node \
    --wait-period=<how often to check in seconds> \
    --stake-amount=<amount to stake, if 0 will restake the last seen amount>
```

Note, right now uses `validator_key.json` to sign transactions as well. It's TODO to add option for a different signing key.

Additionally:
 - Author for Block and Chunk RPCs (Fixes #1121).
 - Validators RPC now shows returns `num_expected_blocks`, `num_produced_blocks` and `shards` for each validator (Fixes #1637):
```
http post http://localhost:3030 jsonrpc=2.0 method=validators id=123 params:='[null]'
{
    "id": "123",
    "jsonrpc": "2.0",
    "result": {
        "current_fishermen": [],
        "current_proposals": [],
        "current_validators": [
            {
                "account_id": "test.near",
                "is_slashed": false,
                "num_expected_blocks": 38,
                "num_produced_blocks": 38,
                "public_key": "ed25519:22skMptHjFWNyuEWY22ftn2AbLPSYpmYwGJRGwpNHbTV",
                "shards": [
                    0
                ],
                "stake": "50445149127868051176979985"
            }
        ],
        "next_fishermen": [],
        "next_validators": [
            {
                "account_id": "test.near",
                "public_key": "ed25519:22skMptHjFWNyuEWY22ftn2AbLPSYpmYwGJRGwpNHbTV",
                "shards": [
                    0
                ],
                "stake": "50445324683980935237733345"
            }
        ]
    }
}
```
 - Network info additionally shows PeerId for `known_producers`.

CC @frol about RPC changes and @amgando for documentation changes of RPC and Restaking itself.

### CODE REVIEW

File: chain/client/src/view_client.rs [hunk 1]
```diff
@@ -281,7 +292,15 @@ impl Handler<GetBlock> for ViewClientActor {
...
-        .map(|block| block.into())
+        .and_then(|block| {
+            self.runtime_adapter
+                .get_block_producer(
+                    &block.header.inner_lite.epoch_id,
+                    block.header.inner_lite.height,
+                )
+                .map(|author| (author, block))
+        })
+        .map(|(author, block)| BlockView::from_author_block(author, block))
```
Comment (bowenwang1996) [inline chain/client/src/view_client.rs]: This can be combined with `.map(|author| (author, block))`

File: chain/client/src/view_client.rs [hunk 2]
```diff
@@ -307,7 +326,21 @@ impl Handler<GetChunk> for ViewClientActor {
...
+        .and_then(|(epoch_id, chunk)| {
+            self.runtime_adapter
+                .get_chunk_producer(
+                    &epoch_id,
+                    chunk.header.inner.height_created,
+                    chunk.header.inner.shard_id,
+                )
+                .map(|author| (author, chunk))
+        })
+        .map(|(author, chunk)| ChunkView::from_author_chunk(author, chunk))
```
Comment (bowenwang1996) [inline chain/client/src/view_client.rs]: Same here

**Comment by amgando** [comment]
appended re-staking service to validator section 

image

**Comment by amgando** [comment]
appended new validator RPC call to RPC API section

image

**Comment by amgando** [comment]
not sure why i'm getting a different response from TestNet

expected: see example above (https://github.com/nearprotocol/nearcore/pull/1962#issue-363335696)

actual: 

```json
{
    "id": "dontcare",
    "jsonrpc": "2.0",
    "result": {
        "current_fishermen": [],
        "current_proposals": [],
        "current_validators": [
            {
                "account_id": "far",
                "is_slashed": false,
                "num_missing_blocks": 0,
                "stake": "200663637577332478904026936259"
            }
        ],
        "next_fishermen": [],
        "next_validators": [
            {
                "account_id": "far",
                "amount": "200665040940372677611389132209",
                "public_key": "ed25519:7rNEmDbkn8grQREdTt3PWhR1phNtsqJdgfV26XdR35QL"
            }
        ]
    }
}
```

**Comment by bowenwang1996** [comment]
well there is a difference between localhost and testnet.

**Comment by ilblackdragon** [comment]
@amgando because TestNet is running older version of the code.

**Pull request merged**